src/libNetwork/P2PComm.cpp

```diff
@@ -422,130 +451,136 @@ void P2PComm::HandleAcceptedConnection(int cli_sock, Peer from)
-    {
-        // Unexpected start byte. Drop this message
-        LOG_GENERAL(WARNING, "Header length or type wrong.");
-    }
-}
-
-void P2PComm::HandleAcceptedConnectionNormal(int cli_sock, Peer from,
-                                             uint32_t message_length,
-                                             SocketCloser cli_sock_closer)
-{
-    vector<unsigned char> message;
+    const uint32_t messageLength = (message[2] << 24) + (message[3] << 16)
+        + (message[4] << 8) + message[5];
 
-    if (!readMsg(&message, cli_sock, from, message_length))
+    // Check for length consistency
+    if (messageLength != message.size() - HDR_LEN)
     {
+        LOG_GENERAL(WARNING, "Incorrect message length.");
         return;
     }
 
-    cli_sock_closer.reset(); // close socket now so it can be reused
-
-    m_dispatcher(message, from);
-}
-
-void P2PComm::HandleAcceptedConnectionBroadcast(int cli_sock, Peer from,
-                                                uint32_t message_length,
-                                                SocketCloser cli_sock_closer)
-{
-    vector<unsigned char> msg_hash;
+    int fd = bufferevent_getfd(bev);
+    struct sockaddr_in cli_addr;
+    socklen_t addr_size = sizeof(struct sockaddr_in);
+    getpeername(fd, (struct sockaddr*)&cli_addr, &addr_size);
+    Peer from(cli_addr.sin_addr.s_addr, cli_addr.sin_port);
 
-    if (!readMsg(&msg_hash, cli_sock, from, HASH_LEN))
+    if (startByte == START_BYTE_BROADCAST)
     {
-        return;
-    }
+        if ((messageLength - HDR_LEN) <= HASH_LEN)
+        {
+            LOG_GENERAL(
+                WARNING,
+                "Hash missing or empty broadcast message (messageLength = "
+                    << messageLength << ")");
+            return;
+        }
 
-    // Check if this message has been received before
-    vector<unsigned char> message;
-    bool found = false;
-    {
-        lock_guard<mutex> guard(P2PComm::GetInstance().m_broadcastHashesMutex);
+        vector<unsigned char> msg_hash(message.begin() + HDR_LEN,
+                                       message.begin() + HDR_LEN + HASH_LEN);
 
-        found = (P2PComm::GetInstance().m_broadcastHashes.find(msg_hash)
-                 != P2PComm::GetInstance().m_broadcastHashes.end());
-        // While we have the lock, we should quickly add the hash
-        if (!found)
+        // Check if this message has been received before
+        bool found = false;
         {
-            if (!readMsg(&message, cli_sock, from, message_length - HASH_LEN))
-            {
-                return;
-            }
-
-            SHA2<HASH_TYPE::HASH_VARIANT_256> sha256;
-            sha256.Update(message);
-            vector<unsigned char> this_msg_hash = sha256.Finalize();
+            lock_guard<mutex> guard(
+                P2PComm::GetInstance().m_broadcastHashesMutex);
 
-            if (this_msg_hash == msg_hash)
+            found = (P2PComm::GetInstance().m_broadcastHashes.find(msg_hash)
```

Comment:
If `P2PComm::GetInstance()` is called multiple times in this function, we can simply declare a variable with a short name to hold the reference.

Reply:
Got it